backend/api/auth.go

```diff
@@ -0,0 +1,44 @@
+package api
+
+import (
+	"context"
+	"errors"
+
+	"github.com/sensu/sensu-go/backend/authentication/jwt"
+	"github.com/sensu/sensu-go/backend/authorization"
+	"github.com/sensu/sensu-go/types"
+)
+
+// addAuthUser finds the user in the provided context, and sets it on the
+// attrs.
+func addAuthUser(ctx context.Context, attrs *authorization.Attributes) error {
+	// Get the claims from the request context
+	claims := jwt.GetClaimsFromContext(ctx)
+	if claims == nil {
+		return errors.New("no claims found in the request context")
+	}
+
+	// Add the user to our request info
+	attrs.User = types.User{
+		Username: claims.Subject,
+		Groups:   claims.Groups,
+	}
+
+	return nil
+}
+
+// authorize is a convenience function for clients that want to test if the
+// operation specified by the attributes is authorized.
+func authorize(ctx context.Context, auth authorization.Authorizer, attrs *authorization.Attributes) error {
+	if err := addAuthUser(ctx, attrs); err != nil {
+		return err
+	}
+	authorized, err := auth.Authorize(ctx, attrs)
+	if err != nil {
+		return err
+	}
+	if !authorized {
+		return authorization.ErrUnauthorized
```

Comment:
The GraphQL service has largely reliant on the errors returned from the controllers interface. This change in the contract will have the trickle down of returning a lot of service errors and exceptions in the web app. See: https://github.com/sensu/sensu-go/blob/5a8ed856967ef3ee4dc0ffdef2f27eadee0a2d8c/backend/apid/graphql/dataloader.go#L336-L360

Reply:
Thanks James, I'll work on getting the error handling mechanisms playing nicely with the new clients.

Reply:
I've reworked these error handlers to use the equivalent errors from the `authorization` and `store` packages.